deploy/kubernetes/overlays/windows/base/kustomization.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```

Comment:
So with one overlay, we have both windows and linux together. What if user need to deploy with mixed OS and versions, e.g, linux stable, and windows dev etc? Or just because some versions are only available in linux, not windows?

Reply:
I think we'll still need to have different overlays for dev/alpha/staging/stable and we can control the versions used in each. For example, the dev overlay could use stable versions for linux, and unstable versions for windows. And in terms of development flow, we can add Windows first to the dev overlay, then once we're ready, add it to staging, and then when we're ready to release, add it to stable.

Reply:
Maybe I miss something here, so you think user should not need a mixed version for windows and linux. If they want to deploy stable version, it will be stable version for both linux and windows? But what if windows don't have stable version yet, user must to deploy dev version for both linux and windows, instead of one stable version and one dev version?

Reply:
Also in the case user already deploy. a Linux driver, and want to add a windows driver, with one overlay, it will try to deploy both linux and windows. Maybe some setup might be messed up with different versions of deployment?

Reply:
I have been thinking the structure again last night :), and realized this is more like a multi-stage process. So instead of base/overlay structure, I propose to use the following as multiple stages and use them as bases. I am working on this structure now and will update. Please let me know what you think about this structure.

- cluster_controller_setup : includes cluster setup and controller yaml file
- node_setup: 
      *  Linux: node and other related yaml files for Linux
      *  Windows:    node and other related yaml files for Windows

- image_update
      *  alpha:
      *. stable:
      *. staging:
      * dev:
...

Reply:
I'm not quite sure how this would work out but am curious to see. One thing to note is that alpha/staging/stable is not necessarily just different versions. Different features may be added depending on their maturity.

>  what if windows don't have stable version yet, user must deploy dev version for both linux and windows, instead of one stable version and one dev version?

In terms of looking at the driver itself, I think we should consider both Linux + Windows parts of the driver to be part of one complete driver instead of 2 independent drivers. So considering that Windows is still alpha right now, I think it is fine if it uses the Linux alpha versions as well (which currently don't have any difference from stable). I don't think a user will want to have a mixed Linux + Windows cluster where the Linux part is stable and the Windows part is alpha. I would expect a user to have 2 separate clusters: a Linux-only cluster used for production, and an experimental Windows-only cluster.

> Also in the case user already deploy a Linux driver, and want to add a windows driver (might be a different version from Linux), with one overlay, it will try to deploy both linux and windows. Some setup might be messed up with different versions of deployment?

I agree with one deployment and one place to define all the versions, it's easier to make sure the versions are in-sync with each other.